### DISCUSSION TITLE
RF Command Duration RFRaw

### DISCUSSION THREAD

**Discussion by artlazza** [arendst/Tasmota#20313] [Q&A]
Hi, I'm using RfRaw to mimic a RF remote control to my Lego lights. I'm using the Sonoff RF Bridge with Tasmota and Portisch Firmware.

I was already able to get the B0 code using RfRaw 177 as below:

_AA B0 29 03 08 04F6 0190 2EE0 281818181818181818181818181818181909090819081819081818190819090909 55_

However, what I noticed is when I send this command:

_RfRaw AA B0 29 03 08 04F6 0190 2EE0 281818181818181818181818181818181909090819081819081818190819090909 55_

The lights turn on, then off. This is exact what happens if I keep the RF remote button pressed (instead of pushing and let it go), the lights keep turning on and off, maybe every half a second. So I'm guessing the command is being sent by the Sonoff RF Bridge for "too long". Is there a way I can configure a shorter pulse/duration?

Thanks!

**Comment by barbudor** [comment]
Frames sent are likely to be repeated to insure proper reception. A receiving device should handle repeating and discard the same message received in a short time.

I can't find this repetition in Tasmota code but Portish seems to have a macro defined to 8
https://github.com/Portisch/RF-Bridge-EFM8BB1/blob/af1bddb3d81c79d67063184219ec21f8249dffd0/inc/RF_Protocols.h#L34

I can't find a way to configure that from Tasmota. 
If you want to change this behavior you need to recompile your own Portish firmware. 
I have no experience with that but the Keil 8051 compiler seems to be provided in Portish repo

**Reply by artlazza** [reply]
I appreciate your reply, I compiled the hex file myself on Simplicity Studio however even changing the RF_TRANSMIT_REPEATS to as low as 1 the behavior is still the same, any chance it's still somewhere else in the firmware (either Tasmota or Portisch)? I noticed that the RF Transmit Led still blinks for about the same time, even after changing the parameter.

**Comment by kainze** [comment]
did you solve the Problem somehow? 
i have a dooya awning with led lights,
Opening and closing works without any problems, but the LED light always switches twice.
